tests/notnull_tests.cpp

```diff
@@ -80,20 +80,20 @@ SUITE(NotNullTests)
 	p = q; // not possible to implicitly convert MyBase* to MyDerived*
 
         not_null<Unrelated*> r = p;
-        not_null<Unrelated*> s = reinterpret_cast<Unrelated*>(p);
+        not_null<Unrelated*> s(reinterpret_cast<Unrelated*>(p));
 #endif
-        not_null<Unrelated*> t = reinterpret_cast<Unrelated*>(p.get());
+        not_null<Unrelated*> t(reinterpret_cast<Unrelated*>(p.get()));
         CHECK((void*)p.get() == (void*)t.get());
     }
 
     TEST(TestNotNullAssignment)
     {
         int i = 12;
-        not_null<int*> p = &i; 
+        not_null<int*> p(&i); 
```

Comment:
If we are getting rid of the assignment operator of T, do we need the test at all? I believe there are already tests for construction.